cmd/tdex/main.go

```diff
@@ -195,9 +201,57 @@ func getClientConn() (*grpc.ClientConn, error) {
 	if !ok {
 		return nil, errors.New("set rpcserver with `config set rpcserver`")
 	}
-	//macaroon := state["macaroon"]
 
-	opts := []grpc.DialOption{grpc.WithDefaultCallOptions(maxMsgRecvSize), grpc.WithInsecure()}
+	opts := []grpc.DialOption{grpc.WithDefaultCallOptions(maxMsgRecvSize)}
+
+	noMacaroons, _ := strconv.ParseBool(state["no_macaroons"])
+	if noMacaroons {
+		opts = append(opts, grpc.WithInsecure())
+	} else {
+		// Load TLS cert for operator interface (enabled automatically when using
+		// macaroon auth)
+		certDatadir, ok := state["tls_cert_path"]
+		if !ok {
+			return nil, fmt.Errorf(
+				"TLS certificate path is missing. Try " +
+					"'tdex config set tls_cert_path /some/dir/including/certificate'",
+			)
+		}
+		certPath := filepath.Join(certDatadir, tlsCertFile)
+
+		tlsCreds, err := credentials.NewClientTLSFromFile(certPath, "")
+		if err != nil {
+			return nil, fmt.Errorf("could not read TLS certificate:  %s", err)
+		}
+		opts = append(opts, grpc.WithTransportCredentials(tlsCreds))
+
+		// Load macaroons and add credentials to dialer
+		if !skipMacaroon {
```

Comment:
whats difference between nomacaroons and skip

Reply:
the trade interface isn't dependent from the macaroon authentication. `skipMacaroons` is basically used by serveMux to distinguish between the trade and the operator interface. For the former macaroon is always disabled, while for the latter it depends on the configuration.